safehttp/file.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp
+
+import (
+	"mime"
+	"net/http"
+	"path/filepath"
+)
+
+// FileResponse is a file response. It is a wrapper around the http.FileServer.
+type FileResponse struct {
+	Root    http.FileSystem
+	Request *IncomingRequest
+}
+
+// ContentType returns the Content-Type based solely on the extension of the
+// filepath. If none is found, the Content-Type is set to
+// "application/octet-stream; charset=binary".
+func (fs *FileResponse) ContentType() string {
+	ct := mime.TypeByExtension(filepath.Ext(fs.Request.URL.Path()))
+	if ct == "" {
+		return "application/octet-stream; charset=binary"
+	}
+	return ct
+}
+
+// Writes writes the response. Write does not override the Content-Type when
+// successfuly serving files. It does override it for redirects and errors.
+//
+// To safely set the Content-Type based on the filepath extension, set the
+// header to the result of ContentType, before calling Write.
+func (fs *FileResponse) Write(rw http.ResponseWriter) {
```

Comment:
JSONResponse returns an error, this does not, is it intended?

I think it would be nice to have a "serializable" interface (unexported or exported) to treat all safe responses the same way.

Also, I would call these "Serialize", not "Write". In the stdlib the "Write" method is defined on stuff you can **write to**. These instead write to the argument, so it feels weird. (Yeah, I know WriteHeader does exactly this but bear with me pls).